packages/protoplugin-test/src/custom-options.test.ts

```diff
@@ -0,0 +1,242 @@
+import { FooMessage } from "./gen/proto/custom_options_pb.js";
+
+describe("custom options", function () {
+  const descriptorSet = getDescriptorSet();
+  describe("finds options correctly", function () {
+    test("file options", () => {
+      for (const file of descriptorSet.files) {
+        if (file.name === "proto/custom_options") {
+          expect(
+            findCustomScalarOption(file, 50000, ScalarType.STRING)
+          ).toEqual("Hello");
+        }
+      }
+    });
+    test("enum options", () => {
+      const enumeration = descriptorSet.enums.get("example.FooEnum");
+      assert(enumeration);
+      expect(
+        findCustomScalarOption(enumeration, 50004, ScalarType.BOOL)
+      ).toBeTruthy();
+    });
+    test("enum value options", () => {
+      const enumeration = descriptorSet.enums.get("example.FooEnum");
+      assert(enumeration);
+      for (const enumValue of enumeration.values) {
+        if (enumValue.name === "OFF") {
+          expect(
+            findCustomScalarOption(enumValue, 50005, ScalarType.UINT32)
+          ).toEqual(321);
+        } else {
+          expect(
+            findCustomScalarOption(enumValue, 50005, ScalarType.UINT32)
+          ).toBeUndefined();
+        }
+      }
+    });
+    test("message options", () => {
+      const msg = descriptorSet.messages.get("example.FooMessage");
+      assert(msg);
+      expect(findCustomScalarOption(msg, 50001, ScalarType.INT32)).toEqual(
+        1234
+      );
+    });
+    test("field options", () => {
+      const msg = descriptorSet.messages.get("example.FooMessage");
+      assert(msg);
+      for (const member of msg.members) {
+        switch (member.kind) {
+          case "oneof":
+            expect(
+              findCustomScalarOption(member, 50003, ScalarType.INT64)
+            ).toEqual(BigInt(42));
+            break;
+          default: {
+            const val = findCustomScalarOption(member, 50002, ScalarType.FLOAT);
+            if (member.name === "foo") {
+              expect(val).toEqual(4.5);
+            } else {
+              expect(val).toBeUndefined();
+            }
+            break;
+          }
+        }
+      }
+    });
+    test("service options", () => {
+      const service = descriptorSet.services.get("example.FooService");
+      assert(service);
+      expect(findCustomEnumOption(service, 50006)).toEqual(1);
+    });
+    test("method options", () => {
+      const service = descriptorSet.services.get("example.FooService");
+      assert(service);
+      for (const method of service.methods) {
+        const option = findCustomMessageOption(method, 50007, FooMessage);
+        expect(option?.foo).toEqual(567);
+        expect(option?.bar).toEqual("Some string");
+        expect(option?.qux?.case).toEqual("quux");
+        expect(option?.qux?.value).toEqual("Oneof string");
+        expect(option?.many).toEqual(["a", "b", "c"]);
+        expect(option?.mapping).toEqual({ testKey: "testVal" });
```

Comment:
@timostamm If it's OK with you, I'd like to leave these additional expectations here just as examples for how to read values from a message option.  If you feel strongly about it, I can remove though.